MvvmCross/Core/MvxSetup.cs

```diff
@@ -50,11 +53,44 @@ public static IMvxSetup Instance()
             return instance;
         }
 
-        protected abstract IMvxApplication CreateApp();
+        protected virtual IMvxViewPresenter ViewPresenter
+        {
+            get
+            {
+                return _presenter;
+            }
+        }
+
+        protected virtual void RegisterImplementations()
+        {
+            // Register Singletons that can be lazy loaded
+            Mvx.IoCProvider.LazyConstructAndRegisterSingleton<IMvxSettings, MvxSettings>();
+            Mvx.IoCProvider.LazyConstructAndRegisterSingleton<IMvxPluginManager, MvxPluginManager>();
+            Mvx.IoCProvider.LazyConstructAndRegisterSingleton<IMvxViewModelLoader, MvxViewModelLoader>();
+            Mvx.IoCProvider.LazyConstructAndRegisterSingleton<IMvxNavigationService, MvxNavigationService>();
+            Mvx.IoCProvider.LazyConstructAndRegisterSingleton<IMvxNavigationSerializer, MvxStringDictionaryNavigationSerializer>();
+            Mvx.IoCProvider.LazyConstructAndRegisterSingleton<IMvxViewModelTypeFinder, MvxViewModelViewTypeFinder>();
+            Mvx.IoCProvider.LazyConstructAndRegisterSingleton<IMvxTypeToTypeLookupBuilder, MvxViewModelViewLookupBuilder>();
+            Mvx.IoCProvider.LazyConstructAndRegisterSingleton<IMvxCommandCollectionBuilder, MvxCommandCollectionBuilder>();
+            Mvx.IoCProvider.LazyConstructAndRegisterSingleton<IMvxChildViewModelCache, MvxChildViewModelCache>();
+            Mvx.IoCProvider.LazyConstructAndRegisterSingleton<IMvxStringToTypeParser, MvxStringToTypeParser>();
```

Comment:
I kind of prefer to have this registration lines splitted in separate methods (as it is now). That way the behavior is much easier to override and the user customizes only what he/she needs.

Reply:
The point of them being lazy is that a developer wanting to override one or more should still call the base method. Then they just register the implementations they want, which will replace the prior registration